sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/IterableConfigProperties.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.autoconfigure.spi;
+
+import io.opentelemetry.api.internal.ConfigUtil;
+import java.util.Iterator;
+import java.util.Set;
+
+/** Config properties that allow access to the property names. */
+public interface IterableConfigProperties extends ConfigProperties, Iterable<String> {
+
+  /**
+   * Returns all property names in the normalized form. System properties are normalized using
+   * {@link ConfigUtil#normalizePropertyKey(String)}. Environment variables are normalized using
+   * {@link ConfigUtil#normalizeEnvironmentVariableKey(String)}.
+   *
+   * @return all property names in the normalized form.
+   */
+  Set<String> getNormalizedPropertyNames();
+
+  @Override
+  default Iterator<String> iterator() {
```

Comment:
I agree with Jack's comment above, but I also believe that java's `Iterator` is a poor api choice that adds little value here. If the use case in the description is a primary driver, I'd like to see methods like these added to the `ConfigProperties` interface instead:

* `Set<String> getNormalizedPropertyNames()` (above)
* `boolean containsProperty(String key)`
* `boolean isEnabled(String key)`
* `boolean isDisabled(String key)`

Reply:
I don't think I agree with these:
```
boolean containsProperty(String key)
boolean isEnabled(String key)
boolean isDisabled(String key)
```
-- it might a bit too much API creep.
But the `Set<String> getNormalizedPropertyNames()` could be a welcome addition.

Reply:
Sure, I appreciate that push back on surface area...but you see where I'm coming from, right? 

If common usages of getting the normalized property names just results in the caller immediately calling `.contains()` then we should encapsulate that and not promote law of Demeter violations. Could definitely be a follow-up PR and doesn't need to stop this one.

Reply:
I think I'll take out the iterable again - it's too surprising